play.pokemonshowdown.com/src/client-main.ts

```diff
@@ -1077,6 +1087,17 @@ export const PS = new class extends PSModel {
 			}, true);
 		}
 
+		/*
+		* Create rooms before /autojoin is sent to the server
+		*/
+		let autojoin = this.prefs.autojoin;
+		if (autojoin) {
+			let rooms = autojoin[this.server.id] || '';
+			rooms.split(",").forEach(title => {
+				this.addRoom({ id: String(toID(title)) as RoomID, title, connected: true }, true);
```

Comment:
```suggestion
				this.addRoom({ id: toID(title) as RoomID, title, connected: true }, true);
```

The `String` cast isn't necessary.

Reply:
i get this if i dont cast it to string
`Conversion of type 'ID' to type 'RoomID' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.`

Reply:
guess i should just convert it to unknown